### PR TITLE
Refactor bigquery to bigtable template to be more extensible

### PR DESCRIPTION
**Pull request by Polber**

### CODE REVIEW

File: v2/bigquery-to-bigtable/pom.xml [hunk 1]
```diff
@@ -51,5 +51,11 @@
                 </exclusion>
             </exclusions>
         </dependency>
+        <dependency>
+            <groupId>com.google.cloud.teleport.v2</groupId>
+            <artifactId>bigtable-common</artifactId>
+            <version>1.0-SNAPSHOT</version>
```
Comment (bvolpato) [inline v2/bigquery-to-bigtable/pom.xml:57]: nit 
```suggestion
            <version>${project.version}</version>
```

File: v2/bigquery-to-bigtable/src/test/java/com/google/cloud/teleport/v2/templates/BigQueryToBigtableIT.java [hunk 2]
```diff
@@ -143,4 +129,66 @@ public void testBigQueryToBigtable() throws IOException {
...
+  public void testBigQueryToBigtableWithQuery() throws IOException {
+    String tableName = "test_table";
+    String colFamily = "names";
+
+    PipelineLauncher.LaunchConfig.Builder options =
+        PipelineLauncher.LaunchConfig.builder(testName, specPath)
+            .addParameter(
+                READ_QUERY,
+                "SELECT * FROM `"
+                    + String.join(".", PROJECT, bigQueryClient.getDatasetId(), tableName)
```
Comment (bvolpato) [inline v2/bigquery-to-bigtable/src/test/java/com/google/cloud/teleport/v2/templates/BigQueryToBigtableIT.java:143]: nit: this `String.join` could be done in a utility method like we have `toTableSpec(table)`

**Review by bvolpato — APPROVED** [review]
This looks great, thanks Jeff!

File: v2/bigquery-to-bigtable/pom.xml (hunk 1 above)
Comment (Polber) [reply v2/bigquery-to-bigtable/pom.xml:57]: Fixed.

**Review by bvolpato — DISMISSED** [review]

File: it/src/main/java/com/google/cloud/teleport/it/matchers/RecordsSubject.java [hunk 3]
```diff
@@ -172,4 +192,172 @@ public void allMatch(Map<String, Object> record) {
...
+    } catch (Exception e) {
+      throw new RuntimeException("Error converting BigQuery Row to Map", e);
+    }
+  }
+
+  /**
+   * Convert BigQuery {@link InsertAllRequest.RowToInsert} to a list of maps.
+   *
+   * @param rows BigQuery rows to parse.
+   * @param excludeCols BigQuery rows to filter out of result.
```
Comment (bvolpato) [inline it/src/main/java/com/google/cloud/teleport/it/matchers/RecordsSubject.java:289]: improve the param doc

File: it/src/main/java/com/google/cloud/teleport/it/matchers/RecordsSubject.java [hunk 4]
```diff
@@ -172,4 +192,172 @@ public void allMatch(Map<String, Object> record) {
...
+   * @return List of maps to use in {@link RecordsSubject}
+   */
+  public static List<Map<String, Object>> bigQueryRowsToRecords(
+      List<InsertAllRequest.RowToInsert> rows, List<String> excludeCols) {
+    List<Map<String, Object>> records = bigQueryRowsToRecords(rows);
+    try {
+      excludeCols.forEach(col -> records.forEach(row -> row.remove(col)));
```
Comment (bvolpato) [inline it/src/main/java/com/google/cloud/teleport/it/matchers/RecordsSubject.java:296]: could be a separate method

File: v2/bigquery-to-bigtable/src/test/java/com/google/cloud/teleport/v2/templates/BigQueryToBigtableIT.java [hunk 5]
```diff
@@ -102,6 +104,7 @@ private void simpleBigQueryToBigtableTest(PipelineLauncher.LaunchConfig.Builder
     bigQueryClient.write(tableName, bigQueryRows);
 
     String colFamily = options.getParameter(WRITE_COL_FAMILY);
+    assert colFamily != null;
```
Comment (bvolpato) [inline v2/bigquery-to-bigtable/src/test/java/com/google/cloud/teleport/v2/templates/BigQueryToBigtableIT.java:107]: use Truth.assertThat

**Review by bvolpato — DISMISSED** [review]
LGTM, a few nits only

File: it/src/main/java/com/google/cloud/teleport/it/matchers/RecordsSubject.java (hunk 3 above)
Comment (Polber) [reply it/src/main/java/com/google/cloud/teleport/it/matchers/RecordsSubject.java:289]: Done.

File: v2/bigquery-to-bigtable/src/test/java/com/google/cloud/teleport/v2/templates/BigQueryToBigtableIT.java (hunk 5 above)
Comment (Polber) [reply v2/bigquery-to-bigtable/src/test/java/com/google/cloud/teleport/v2/templates/BigQueryToBigtableIT.java:107]: Done.

File: it/src/main/java/com/google/cloud/teleport/it/matchers/RecordsSubject.java (hunk 4 above)
Comment (Polber) [reply it/src/main/java/com/google/cloud/teleport/it/matchers/RecordsSubject.java:296]: I will separate this out in a future PR.

File: v2/bigquery-to-bigtable/src/test/java/com/google/cloud/teleport/v2/templates/BigQueryToBigtableIT.java (hunk 2 above)
Comment (Polber) [reply v2/bigquery-to-bigtable/src/test/java/com/google/cloud/teleport/v2/templates/BigQueryToBigtableIT.java:135]: Done.

File: it/src/main/java/com/google/cloud/teleport/it/matchers/RecordsSubject.java (hunk 4 above)
Comment (Polber) [reply it/src/main/java/com/google/cloud/teleport/it/matchers/RecordsSubject.java:306]: I will separate this out in a future PR.

**Pull request merged**